R/download_models.R

```diff
@@ -0,0 +1,50 @@
+#' Download SS3 test models
+#'  
+#' Function to download the models folder within the 
+#' nmfs-stock-synthesis/test-models repsitory
+#' @param dir The folder where the models subfolder should be written to.
+#' @param branch The name of the github branch to download.
+#' @template overwrite
+#' @return Invisibly return a value revealing whether the files were copied 
+#'  (TRUE) or not (FALSE). This function is used for its side effects of
+#'  downling SS3 test models.
+#' @examples
+#'   download_models(dir = getwd())
+#'   model_name <- list.files("models") # see the model names
+#'   # remove files
+#'   unlink(file.path("models"), recursive = TRUE)
+#' @author Kathryn Doering
+#' @references [nmfs-stock-synthesis/test-models repository](https://github.com/nmfs-stock-synthesis/test-models#test-models)
+#' @export
+download_models <- function(dir = file.path("inst", "extdata"), 
+  branch = "main", overwrite = FALSE) {
+  # checks
+  if (!dir.exists(dir)) {
+    stop("dir ", dir, " does not exist")
+  }
+  zip_file_path <- file.path(dir, "test-models.zip")
+  result <- tryCatch(utils::download.file(
```

Comment:
I would prefer to change 26-33 to
```
  result <- tryCatch(
    utils::download.file(
      url = paste0(
        "https://github.com/nmfs-stock-synthesis/test-models/archive/",
        branch,
        ".zip"
        ),
      destfile = zip_file_path
    ),
    warning = function(e) {
      stop(
        "The test-models zip file could not be downloaded.",
        " Does the branch (", branch, ") exist?",
        call. = FALSE
      )
    }
  )
```
where the indents make it more clear what `tryCatch` is doing and the behavior of the warning is linked to the `stop()`.

Reply:
So, I tried this out, and when using warning= instead of error= it creates a situation where the zip file connection is left open and so isn't properly deleted (because it is being accessed by a different program). This then means the tests can't be fully cleaned up. I found changing to `error=` prevents this unintended consequence from happening; Do you think that is an acceptable compromise?

Reply:
I think that should be fine.

Reply:
Thanks, I will push my changes shortly so you can see them!